libxslt fails to cross compile

### Steps To Reproduce
Steps to reproduce the behavior (on "staging" at time of writing):
1. `nix build .#pkgsCross.aarch64-multiplatform.libxslt`

### Build log
```
...
configure: error: no suitable Python interpreter found
```

--- a/pkgs/development/libraries/libxslt/default.nix
+++ b/pkgs/development/libraries/libxslt/default.nix
@@ -50,10 +50,7 @@ stdenv.mkDerivation rec {
     "--without-debug"
     "--without-mem-debug"
     "--without-debugger"
-  ] ++ lib.optionals pythonSupport [
-    "--with-python=${python}"
-  ] ++ lib.optionals (!pythonSupport) [
-    "--without-python"
+    (lib.withFeatureAs pythonSupport "python" python)
   ] ++ lib.optionals (!cryptoSupport) [
     "--without-crypto"
   ];

--- a/pkgs/development/libraries/libxslt/default.nix
+++ b/pkgs/development/libraries/libxslt/default.nix
@@ -52,6 +52,8 @@ stdenv.mkDerivation rec {
     "--without-debugger"
   ] ++ lib.optionals pythonSupport [
     "--with-python=${python}"
+  ] ++ lib.optionals (!pythonSupport) [
+    "--without-python"
   ] ++ lib.optionals (!cryptoSupport) [
     "--without-crypto"
   ];

